### PR TITLE
MOTOR-1169 Update PyMongo Deps and Remove Workarounds

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
evergreen retry

**Comment by blink1073** [comment]
evergreen retry

**Review by Jibola — APPROVED** [review]
So far so good! I did quick grep of the codebase to see if there were any missed sites, to run by you: 
- [A `hasttr(ClientEncryption, "encrypt_expression")` check in `core.py`](https://github.com/blink1073/motor/blob/b6522a9b01712b105d431773df47092de0b24d20/motor/core.py#L2106)

Here's the only one I found. If that's not relevant, let me know. Otherwise, LGTM.

**Comment by blink1073** [comment]
Good catch!  That was added in 4.4.

**Review by Jibola — APPROVED** [review]
LGTM!

File: doc/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,22 @@ Changelog
...
+- Added new helper methods for Atlas Search Index (requires MongoDB Server 7.0+):
+  :meth:`~motor.motor_tornado.MotorCollection.list_search_indexes`,
+  :meth:`~motor.motor_tornado.MotorCollection.create_search_index`,
+  :meth:`~motor.motor_tornado.MotorCollection.create_search_indexes`,
+  :meth:`~motor.motor_tornado.MotorCollection.drop_search_index`,
+  :meth:`~motor.motor_tornado.MotorCollection.update_search_index`
+- Added :meth:`~motor.motor_tornado.MotorDatabase.cursor_command`
+  and :meth:`~motor.motor_tornado.MotorCommandCursor.try_next` to support
+  executing an arbitrary command that returns a cursor.
+
```
Comment (Jibola) [inline doc/changelog.rst:21]: Should this also include the change message for `encrypt_expression`?

**Review by Jibola — CHANGES_REQUESTED** [review]
Whoops! One commentary left.

File: doc/changelog.rst (hunk 1 above)
Comment (blink1073) [reply doc/changelog.rst:21]: It is in the 3.2 release notes.  :)

**Review by Jibola — APPROVED** [review]

**Pull request merged**